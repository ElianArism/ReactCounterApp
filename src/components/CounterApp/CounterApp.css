p {
  margin-bottom: 40px;
  margin-left: 10px;
}

button {
  border: none;
  border-radius: 10px;
  padding: 10px;
  margin: 0 10px;
  background-color: #31364b;
  color: #f3f3f3;
  cursor: pointer;
}

button:first-of-type {
  margin-left: 0;
}