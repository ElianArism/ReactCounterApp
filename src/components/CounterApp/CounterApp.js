import PropTypes from 'prop-types'
import {useState} from 'react'
import './CounterApp.css'

const CounterApp = ({value}) => {
  const [counter, setCounter] = useState( value ) 

  const add = (e) => {
    setCounter(counter + 1) // || setCounter( (counter) => counter + 1)
  }

  const substract = () => {
    setCounter(counter - 1)
  }

  const reset = () => {
    setCounter(value)
  }

  return (
    <>
      <h1> Counter App </h1>
      <p>{ counter }</p> 
      
      <button onClick={ add }> Add </button>
      <button onClick={ substract }> Substract </button>
      <button onClick={ reset }> Reset </button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}

export default CounterApp