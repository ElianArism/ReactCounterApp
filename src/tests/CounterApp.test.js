import { shallow } from "enzyme"
import CounterApp from "../components/CounterApp/CounterApp"


describe('CounterApp', () => {
  const wrapper = shallow( <CounterApp value={10} /> ) // Aisla el componente y le toma un snapshot 
  
  beforeEach( () => {
    wrapper.find('button').at(2).simulate('click')
  })

  test('It should prove CounterApp renders correctly', () => {
    expect(wrapper).toMatchSnapshot() // Se encarga de asegurarse que el snapshot y el componente coincidan
  })

  test('It should prove CounterApp renders with 100 as value', () => {
    const wrapper = shallow( <CounterApp value={100} /> ) 
    const paragraph = wrapper.find('p').text() // encuentra un html element en el componente
    
    expect(paragraph).toBe( '100' )
  })

  test('It should increment counter with add button', () => {
    // const btn = wrapper.find('button').at(0).html() // Obt first button tag
    wrapper.find('button').at(0).simulate('click') // Simular click evt
    const counterText = wrapper.find('p').text().trim()

    expect(counterText).toBe('11')
  })

  test('It should decrement counter with substract button', () => {
    wrapper.find('button').at(1).simulate('click') 
    const counterText = wrapper.find('p').text().trim()
    
    expect(counterText).toBe('9')
  })

  test('It should reset counter with reset button', () => {
    wrapper.find('button').at(0).simulate('click') 
    wrapper.find('button').at(0).simulate('click') 
    wrapper.find('button').at(0).simulate('click') 

    wrapper.find('button').at(2).simulate('click') 

    const counterText = wrapper.find('p').text().trim()

    expect(counterText).toBe('10')
  })


})
