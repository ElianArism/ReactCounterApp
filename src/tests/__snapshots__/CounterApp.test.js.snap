// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp It should prove CounterApp renders correctly 1`] = `
<Fragment>
  <h1>
     Counter App 
  </h1>
  <p>
    10
  </p>
  <button
    onClick={[Function]}
  >
     Add 
  </button>
  <button
    onClick={[Function]}
  >
     Substract 
  </button>
  <button
    onClick={[Function]}
  >
     Reset 
  </button>
</Fragment>
`;

